reference_data/management/tests/update_gencode_tests.py

```diff
@@ -0,0 +1,159 @@
+    'chr1	HAVANA	transcript	621059	622053	.	-	.	gene_id "ENSG00000284662.1_2"; transcript_id "ENST00000332831.4_2"; gene_type "protein_coding"; gene_name "OR4F16"; transcript_type "protein_coding"; transcript_name "OR4F16-201"; level 2; protein_id "ENSP00000329982.2"; transcript_support_level "NA"; hgnc_id "HGNC:15079"; tag "basic"; tag "appris_principal_1"; tag "CCDS"; ccdsid "CCDS41221.1"; havana_gene "OTTHUMG00000002581.3_2"; havana_transcript "OTTHUMT00000007334.3_2"; remap_num_mappings 1; remap_status "full_contig"; remap_target_status "overlap";\n',
+    # feature_type is 'CDS'
+    # gene_id not in existing_gene_ids and transcript_size > ...
+    'chr1	HAVANA	CDS	621099	622034	.	-	0	gene_id "ENSG00000284662.1_2"; transcript_id "ENST00000332831.4_2"; gene_type "protein_coding"; gene_name "OR4F16"; transcript_type "protein_coding"; transcript_name "OR4F16-201"; exon_number 1; exon_id "ENSE00002324228.3"; level 2; protein_id "ENSP00000329982.2"; transcript_support_level "NA"; hgnc_id "HGNC:15079"; tag "basic"; tag "appris_principal_1"; tag "CCDS"; ccdsid "CCDS41221.1"; havana_gene "OTTHUMG00000002581.3_2"; havana_transcript "OTTHUMT00000007334.3_2"; remap_original_location "chr1:-:685719-686654"; remap_status "full_contig";\n',
+    # len(record["chrom"]) > 2
+    'GL000193.1	HAVANA	gene	77815	78162	.	+	.	gene_id "ENSG00000279783.1_5"; gene_type "processed_pseudogene"; gene_name "AC018692.2"; level 2; havana_gene "OTTHUMG00000189459.1_5"; remap_status "full_contig"; remap_num_mappings 1; remap_target_status "new";\n',
+]
+
+
+class UpdateGencodeTest(TestCase):
+    fixtures = ['users', 'reference_data']
+    multi_db = True
+
+
+    def setUp(self):
+        # Create a temporary directory
+        self.test_dir = tempfile.mkdtemp()
+        self.temp_file_path = os.path.join(self.test_dir, 'gencode.v31lift37.annotation.gtf.gz')
+        with gzip.open(self.temp_file_path, 'w') as f:
+            f.write(u''.join(GTF_DATA))
+
+    def tearDown(self):
+        # Close the file, the directory will be removed after the test
+        shutil.rmtree(self.test_dir)
+
+    @mock.patch('os.path.isfile')
+    def test_update_gencode_command_arguments(self, mock_isfile):
+        # Test missing test required argument
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode')
+        self.assertEqual(ce.exception.message, 'Error: argument --gencode-release is required')
+
+        # Test required argument out-of-range
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=18')
+        self.assertEqual(ce.exception.message, 'Error: argument --gencode-release: invalid choice: 18 (choose from 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31)')
+
+        # Test genome_version out-of-range
+        with self.assertRaises(CommandError) as ce:
+            call_command('update_gencode', '--gencode-release=19', '/var/tmp', '39')
```

Comment:
you should not use real files in real tmp directories, you should mock out all system calls

Reply:
In these test, the tmp file names in the argument will never be used for real file accesses (exception will happen before a file access). They are used for command argument checking only. Therefore, no needs to mock out file operations for these tests.

Reply:
Thats good then. Can you change it to something more obviously fake, maybe like `/local/path` or `/mock/dir` or something like that?

Reply:
Okay, change to /mock_path/...

Reply:
This still needs to be done

Reply:
Sorry. Updated this time.